internal/node/indexer/server.go

```diff
@@ -201,27 +232,53 @@ func (s *Server) handleTasks(ctx context.Context, tasks *engine.Tasks) error {
 	return nil
 }
 
-func (s *Server) initializeMeter() (err error) {
-	// init ethereum client
-	switch s.config.Network {
-	case network.Arweave:
-		arweaveClient, err := arweave.NewClient()
-		if err != nil {
-			return fmt.Errorf("new arweave client: %w", err)
-		}
+// checkWorkerStatus checks if the worker is ready.
+func (s *Server) checkWorkerStatus(ctx context.Context, checkpoint engine.Checkpoint) (err error) {
+	// Check if the worker is ready
+	var state monitor.CheckpointState
+	if err := json.Unmarshal(checkpoint.State, &state); err != nil {
+		zap.L().Error("unmarshal checkpoint state", zap.Error(err))
+		return err
+	}
 
-		s.arweaveClient = arweaveClient
-	case network.Farcaster:
-		break
-	default:
-		ethereumClient, err := ethereum.Dial(context.Background(), s.config.Endpoint)
-		if err != nil {
-			return fmt.Errorf("dial ethereum: %w", err)
+	// get current indexing block height, number or event id and the latest block height, number, timestamp of network
+	currentWorkerState, latestWorkerState, err := s.getWorkerIndexingStateByClient(ctx, state)
+	if err != nil {
+		zap.L().Error("get latest block height or number", zap.Error(err))
+		return err
+	}
+
+	return s.flagIndexingWorker(ctx, currentWorkerState, latestWorkerState, monitor.NetworkTorlerance[s.config.Network])
+}
+
+// flagIndexingWorker compares the current and latest block height/number. If the difference is less than the tolerance, the worker is flagged as ready, otherwise it is flagged as indexing.
+func (s *Server) flagIndexingWorker(ctx context.Context, currentWorkerState, latestWorkerState, networkTolerance uint64) error {
+	if latestWorkerState-currentWorkerState < networkTolerance {
+		// Cache worker status to Redis.
+		if err := s.updateWorkerStatus(ctx, s.config.Network.String(), s.config.Worker.String(), workerx.StatusReady.String()); err != nil {
+			return fmt.Errorf("cache token metadata: %w", err)
+		}
+	} else {
+		// Cache worker status to Redis.
+		if err := s.updateWorkerStatus(ctx, s.config.Network.String(), s.config.Worker.String(), workerx.StatusIndexing.String()); err != nil {
+			return fmt.Errorf("cache token metadata: %w", err)
 		}
```

Comment:
updateWorkerStatus error message is incorrect or misleading

Reply:
```go
func (s *Server) flagIndexingWorker(ctx context.Context, currentWorkerState, latestWorkerState, networkTolerance uint64) error {
	var status string
	if latestWorkerState-currentWorkerState < networkTolerance {
		status = workerx.StatusReady.String()
	} else {
		status = workerx.StatusIndexing.String()
	}
	return s.updateWorkerStatus(ctx, s.config.Network.String(), s.config.Worker.String(), status)
}
```

remove the duplicate calls to s.updateWorkerStatus by determining the status first and then calling the function once.